python-frontend/src/main/java/org/sonar/python/semantic/v2/FunctionTypeBuilder.java

```diff
@@ -111,6 +118,15 @@ public FunctionTypeBuilder withInstanceMethod(boolean instanceMethod) {
   }
 
   public FunctionTypeBuilder withReturnType(PythonType returnType) {
+    if (returnType instanceof LazyType lazyType) {
+      withReturnType(new LazyTypeWrapper(lazyType));
+    } else {
+      withReturnType(new SimpleTypeWrapper(returnType));
+    }
```

Comment:
Do we need to check this ? If we always go for LazyTypeWrapper, `TypeUtils.resolved` will do the check for us.